### PR TITLE
Add Interceptor Overload for Send

### PR DESCRIPTION
**Pull request by hwoodiwiss**
This PR adds an override to `InterceptingHttpMessageHandler` for the synchronous `Send` method of `DelegatingHandler` to allow for calls to `HttpClient.Send` to be intercepted.

First pass includes some pretty icky sync-over-async 😢 , however, to avoid this would require some significant re-writes for what feels like a bit of an edge-case.

I've added this, as after upgrading a project to .NET 8 with `AWSSDK.DynamoDBv2` versions > `3.7.300`, I had some failing tests due to some AWS calls not being intercepted due to using the synchronous `HttpClient,Send` overload ([here](https://github.com/aws/aws-sdk-net/blob/0d801e904d0cac58455339b63727135de67a65a1/sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs#L486-L511))

### CODE REVIEW

**Comment by martincostello** [comment]
> I've added this, as after upgrading a project to .NET 8 with AWSSDK.DynamoDBv2 versions > 3.7.300, I had some failing tests due to some AWS calls not being intercepted due to using the synchronous HttpClient.Send overload

Ahhhhhhh. So that explains why that wasn't working. I'd noticed the interception wasn't working in one of our internal libraries, but I'd been able to work around it in the short term by disabling it from fetching the table metadata like noticed in this blog post: https://aws.amazon.com/blogs/developer/improved-dynamodb-initialization-patterns-for-the-aws-sdk-for-net/

Mystery solved.

File: src/HttpClientInterception/PublicAPI/net6.0/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+#nullable enable
```
Comment (martincostello) [inline src/HttpClientInterception/PublicAPI/net6.0/PublicAPI.Unshipped.txt:1]: The new bit _should_ go in the unshipped files until it actually gets to NuGet, then it'll get moved post-release.

However that's probably not worth doing if we just ship this on Monday with no other changes.

File: src/HttpClientInterception/InterceptingHttpMessageHandler.cs [hunk 2]
```diff
@@ -41,6 +41,12 @@ public InterceptingHttpMessageHandler(HttpClientInterceptorOptions options, Http
         _options = options ?? throw new ArgumentNullException(nameof(options));
     }
 
+#if NET6_0_OR_GREATER
+    /// <inheritdoc />
+    protected override HttpResponseMessage Send(HttpRequestMessage request, CancellationToken cancellationToken) =>
+        SendAsync(request, cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (martincostello) [inline src/HttpClientInterception/InterceptingHttpMessageHandler.cs:47]: Yeah this is a bit ick, but as it's a test library and probably quite a minority subset of cases (as it's taken 2 years to hit this) this is fine.

**Review by martincostello — COMMENTED** [review]
Let's try out the version from the PR with the internal library on Monday to check it solves the stuff I'd run into as well. If it does then we can do a 4.1(?) or whatever the next minor version will be.

**Comment by hwoodiwiss** [comment]
> Let's try out the version from the PR with the internal library on Monday to check it solves the stuff I'd run into as well. If it does then we can do a 4.1(?) or whatever the next minor version will be.

Sounds good to me, I inherited the interceptor with this change added in the application I ran into this on and it seemed to work, so 🤞🏼

**Comment by martincostello** [comment]
Changes validated internally.
